### PR TITLE
Style and formatting rules

### PR DESCRIPTION
**Pull request by jacobgarder**
Style, formatting can always can be a religious thing, but when we started to discuss it, I just put in my 50 cents 😀.
For me it is no big deal, but I think what ever rules are set, they should as clear as possible and there were some things I found a bit unclear when I started to contribute to this project that I more or less have guessed. 

So Mitch, please correct me if I'm wrong:
- Formatting is done with Black
- Indentation is done with spaces (4) and not tabs (Black)
- Line-length is set to 127 (matching github width)
- Flake8 is used for linting in the pipeline.
- Pylint is used (to some extent)

Apart from this, I also think personally that:
- Imports should be sorted (with isort)
- Import should be absolute (not reative) as this is more clear and I have come across IDEs that don't work very well with them (but that was a long time ago)

In this PR I have included configuration for these formatting tools which makes it easier for all contributors to conform to the formatting rules and updated documentation abit. 

I have also reformatted the files accordingly, even the docsrc and tests directory...this could be argued if they should be included or not..

### CODE REVIEW

File: tests/zia/test_users.py [hunk 1]
```diff
@@ -291,29 +292,6 @@ def test_list_groups_with_max_items_1(zia, paginated_items):
     assert len(resp) == 1
 
 
-@responses.activate
```
[the remark this answers is not in the document]
Comment (mitchos) [reply tests/zia/test_users.py:294]: Actually I think I just forgot to rename this one from users to groups when copying it across. We can see the responses mock URL has the /groups endpoint.

Can you please rename to the below and this file is all good:
`def test_list_groups_with_max_items_150(zia, paginated_items):`

File: pyproject copy.toml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+[tool.poetry]
```
Comment (mitchos) [inline pyproject copy.toml:1]: I'm not sure what the `pyproject copy.toml `file is for? It looks the same as `pyproject.toml` so assuming this was just an accidental include?

**Review by mitchos — CHANGES_REQUESTED** [review]
Can you just review those two comments and I'll approve this after.

Thanks again for your work on this one :smiley:

File: pyproject copy.toml (hunk 2 above)
Comment (jacobgarder) [reply pyproject copy.toml:1]: Yes, that was totally accidental. :)
Removed.

File: tests/zia/test_users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacobgarder) [reply tests/zia/test_users.py:294]: Aha..ok. Will rename it.

**Comment by jacobgarder** [comment]
Thanks! 😃
Added the changes you requested.

**Review by mitchos — APPROVED** [review]
Looks good to me. Happy we're moving forward with this.

**Pull request merged**